plugins/modules/system/java_cert.py

```diff
@@ -167,9 +170,19 @@
 
 import os
 import re
+import tempfile
+import random
+import string
 
 # import module snippets
 from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.crypto.support import load_certificate
```

Comment:
This import comes from another collection that is **not** a dependency of this collection. Please remove this.

Reply:
It's apparently needed, how to import it is the question? 
This is not originally my code, it's only a port because ansible community migrated to collections and wildly closed all PR created in the ansible repository.

Reply:
You will have to find another way to solve this. Using something from community.crypto is not possible, and also you are using a deprecated part of community.crypto which depends on an external library that is not always installed, and you aren't doing any error handling for that part.

I suggest you use the `openssl` command line tool to extract the value you need.

> edit: I rephrase: it is a dependency as from now , the other solution would be to move the java_cert module to the crypto collection which would have make sense imho

It will not become a dependency, as long as the import stays here this PR will not be merged.

Migrating this module to community.crypto only makes sense if someone wants to actively maintain the module there.

Reply:
> I'm however eager to have that feature included in the java_cert module

Do you mean the import statement? Or this PR in general? The current version of the PR no longer uses the import statement.

> I saw what you did with kubevirt which migrated to a separated collection.

The kubevirt stuff depended on community.kubernetes, so it was moved to be able to drop the collection dependency.

> I think I'll repeat myself, and maybe migrating them to the community.crypto would help here. I don't know about the process and if it's allowed by the community to do so.

I already replied to that, did you see my reply above?